test/DataLayouts/data0d.jl

```diff
@@ -0,0 +1,118 @@
+using Test
+using JET
+
+using ClimaCore.DataLayouts
+using StaticArrays
+using ClimaCore.DataLayouts: get_struct, set_struct!
+
+TestFloatTypes = (Float32, Float64)
+
+@testset "F" begin
+    for FT in TestFloatTypes
+        S = Tuple{Complex{FT}, FT}
+        array = rand(FT, 1, 3)
+
+        data = F{S}(array)
+        @test getfield(data, :array) == @view(array[:, :])
+
+        # test tuple assignment
+        data[1] = (Complex{FT}(-1.0, -2.0), FT(-3.0))
```

Comment:
it should act like a `Ref`:
```suggestion
        data[] = (Complex{FT}(-1.0, -2.0), FT(-3.0))
```

Reply:
Thank you! Was unaware of this. There are likely going to be more things like this that I miss as I build this PR.